source/UIAUtils.py

```diff
@@ -119,3 +120,71 @@ def iterUIARangeByUnit(rangeObj,unit):
 	if tempRange.CompareEndpoints(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)<0:
 		tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)
 		yield tempRange.clone()
+
+def getEnclosingElementWithCacheFromUIATextRange(textRange,cacheRequest):

```

Comment:
I think a docstring would be good here. Only has to point out that it uses getEnclosingElementBuildCache or falls back to getting the element and then building the cache if getEnclosingElementBuildCache is not supported.